HW0/P4/P4.py

```diff
@@ -28,49 +28,64 @@
     # Part 1:
     #
     # Load true trajectory and plot it
-    # Normally, this data wouldn't be available in the real world
-    #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    s_true = np.loadtxt('P4_trajectory.txt', delimiter = ',')
+    ax.plot(s_true[:,0], s_true[:,1], s_true[:,2],'--b', label = 'True trajectory')
 
     #####################
     # Part 2:
-    #
     # Read the observation array and plot it (Part 2)
-    #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    s_measure = np.loadtxt('P4_measurements.txt', delimiter= ',')
+    r_matrix = np.matrix([[1/rx,0,0], [0,1/ry,0], [0,0,1/rz]])
+    s_obs = s_measure*r_matrix
+    s_obs = np.squeeze(np.asarray(s_obs))
+    ax.plot(s_obs[:,0], s_obs[:,1], s_obs[:,2],'.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
-    #####################
-
-    # A = ?
-    # a = ?
-    # s = ?
-
+    A = np.matrix([[1,0,0,dt,0,0], [0,1,0,0,dt,0], [0,0,1,0,0,dt], 
```

Comment:
Good! here you could also make use of sparse numpy array, more specifically using np.diag() is more memory efficient and (a little more) readable